source/includes/data-modeling/field-types.rb

```diff
@@ -0,0 +1,251 @@
+# start-define-fields
+class Person
+  include Mongoid::Document
+  field :name, type: String
+  field :date_of_birth, type: Date
+  field :weight, type: Float
+end
+# end-define-fields
+
+# start-define-untyped
+class Person
+  include Mongoid::Document
+
+  field :name, type: String
+  field :preferences
+end
+# end-define-untyped
+
+# start-untyped
+product = Product.new(properties: "color=white,size=large")
+# properties field saved as String: "color=white,size=large"
+
+product = Product.new(properties: {color: "white", size: "large"})
+# properties field saved as Object: {:color=>"white", :size=>"large"}
+# end-untyped
+
+# start-stringified-symbol
+class Post
+  include Mongoid::Document
+  
+  field :status, type: StringifiedSymbol
+end
+  
+# Save status as a symbol
+post = Post.new(status: :hello)
+# status is stored as "hello" on the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+
+# Save status as a string
+post = Post.new(status: "hello")
+# status is stored as "hello" in the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+# end-stringified-symbol
+
+# start-hash
+class Person
+  include Mongoid::Document
+  field :first_name
+  field :url, type: Hash
+  
+  def set_vals
+    self.first_name = 'Daniel'
+    self.url = {'home_page' => 'http://www.homepage.com'}
+    save
+  end
```

Comment:
Q: why does this example use this class method instead of using regular methods to create an instance?

Reply:
Looks like in the original it was made like that to compare with an example that won't work. Updated to use regular methods like the other examples.